crates/ruff_linter/src/rules/fastapi/rules/fastapi_unused_path_parameter.rs

```diff
@@ -136,12 +136,26 @@ pub(crate) fn fastapi_unused_path_parameter(
     // Extract the path parameters from the route path.
     let path_params = PathParamIterator::new(path.to_str());
 
+    let dependables = keywordable_parameters(function_def)
+        .filter_map(|parameter_with_default| {
+            match Dependable::from_parameter(parameter_with_default, checker.semantic()) {
+                Dependable::None => None,
```

Comment:
Considering that you filter out `Dependable::None` here, I'd suggest to change `Dependable::from_paramter` to return an `Option<Dependable>` instead. It has the advantage (other than just being less code) that the compiler can prove, that `Dependable::None` is not a valid value after line 139